custom_components/frigate/media_source.py

```diff
@@ -346,19 +340,22 @@ def get_frigate_server_path(self) -> str:
         # missing attribute.
 
         in_parts = [
-            self.get_identifier_type(),
+            self.get_identifier_type() if not self.camera else "vod",
             self.year_month,
             f"{self.day:02}" if self.day is not None else None,
             f"{self.hour:02}" if self.hour is not None else None,
             self.camera,
-            self.recording_name,
         ]
 
         out_parts = []
         for val in in_parts:
             if val is None:
                 break
             out_parts.append(str(val))
+
+        if self.camera:
+            out_parts.append("index.m3u8")
```

Comment:
I think you can just add this to `in_parts` definition above? Something like:
in_parts = [
...
   "index.m3u8" if self.camera else None
]